lightning/src/routing/scoring.rs

```diff
@@ -117,13 +146,34 @@ impl<'a, T: 'a + Score> LockableScore<'a> for RefCell<T> {
 	}
 }
 
-impl<S: Score, T: DerefMut<Target=S>> Score for T {
-	fn channel_penalty_msat(&self, short_channel_id: u64, send_amt_msat: u64, channel_capacity_msat: Option<u64>, source: &NodeId, target: &NodeId) -> u64 {
-		self.deref().channel_penalty_msat(short_channel_id, send_amt_msat, channel_capacity_msat, source, target)
+#[cfg(c_bindings)]
+/// A concrete implementation of [`LockableScore`] which supports multi-threading.
+pub struct ScoreHolder<S: Score> { // TODO(jkczyz): Name this better
```

Comment:
Hmm... given our implementation of `Score` is called `Scorer`, maybe simply name this `LockableScorer`.

Reply:
Hmm, true, it is still generic over `Score`, though, so is that a misnomer?

Reply:
How about `AsyncLockableScore`?

Reply:
Wait, what is the `Async` in reference to? Maybe `ConcreteLockableScore` or `MultiThreadedLockableScore`?

Reply:
That it implements the locking using a `Mutex`... guess that's really synchronous. :P `MultiThreadedLockableScore` is fine by me.